STYLE_GUIDE.md

```diff
@@ -0,0 +1,75 @@
+# Introduction
+
+This is the code style guide for the FAST tooling repository. When contributing to this project please follow the standards defined in this guide.
+
+## Files
+
+File and folder naming in general should not include special characters or space. Alphanumeric is preffered, separated by dash or underscore depending on the type of file. Names should be as concise as possible while also being meaningful.
+
+### Organization
+
+Folder structure should be as flat as possible. This is not to say that all files should go into a single folder, folders should exist where they make sense for larger categorizing purposes and file names can be leveraged to supply additional context.
+
+Example:
+```text
+my-component/
+└─ my-component.ts
+└─ my-component.style.ts
+└─ my-component.spec.ts
+```
+
+### Scripts
+
+As seen in the above example, script files should be all lowercase and dash separated. If there is need for additional context, marking a test file in the example above, this should be included separating the additional context with a period at the end of the file name.
+
+### Markdown
+
+Markdown files should be all uppercase, separated by an underscore e.g., CODE_OF_CONDUCT.md. This allows them to stand out from the rest of the files as documentation.
+
+## Syntax
+
+Much of the primarily TypeScript syntax is taken care of via `eslint` and prettier. There are times when the linter is unable to determine when naming conventions are broken, these situations are covered in the sections below.
+
+### Functions & Methods
+
+Functions & methods should be camelCase.
+
+Example:
+```ts
+myFunction(): void {
+    // example code
+}
+```
+
+### Web Component Attributes
+
+Web component attributes should be camelCased as properties and be dash separated as the HTML attribute.
+
+Example:
+```ts
+@attr({ attribute: "select-mode" })
+public selectMode: string;
+```
+
+An exception to this is if an attribute matches a standard HTML attribute which may not include a dash.
+
+Example:
+```ts
+@attr({ attribute: "readonly" })
+public readOnly: boolean;
+```
+
+### HTML class names
+
+Class names should follow the BEM guides, where there is a single block and an optional element and modifier. The separator for a block and element should be a hypen and the separator for a block/element and modifier should be two underscores.
```

Comment:
I think you will find that, at least in web components, the selectors are much simpler and you may not need this too much. It's probably worth checking with Chris and Brian to see how they are handling this in the core components.

Reply:
I want to ensure that we are future-proofed and consistent going forward, FAST tooling specifically may have some rather large and complex web components in the future (think the React Form component), though we will try to keep them small. My thinking is that enforcing standard naming that does not cause confusion in a complex component is preferable to attempting to use simpler class names, which might work well for a button component but don't scale.